lib/src/domain/usecase/usecase_interface.dart

```diff
@@ -0,0 +1,7 @@
+import '../../core/parameter/parameter_object.dart';
+import '../../core/util/data_state.dart';
+import '../entity/movie.dart';
+
+abstract interface class IUseCase {
+  Future<DataState<List<Movie>>> call(FetchMovieParams params);
```

Comment:
To make this interface 100% generic, I'd change this for:
```
abstract interface class IUseCase<T,Y> {
  Future<DataState<T> call(Y params)
}
```

Reply:
done